exploration/01_baseline_calc_10yr_doy.py

```diff
@@ -0,0 +1,57 @@
+"""
+This script calculates the DOY baseline for FloodScan raster data at pixel
+level. It is currently in `exploratory` repo as it just uses GDRIVE `.nc` file
+to make local development more effecient (vs loading historical data from blob)
+
+In the official pipeline we can link it to the historical tif record on the
+blob.
+
+Steps:
+
+1. Smooth historical data w/ 10 day rolling mean (centered)
+2. filter data to last 10 years of data (exlcusive of current year)
+3. Calculate the mean SFED value per day of year from smoothed raster stack.
+"""
+
+import datetime
+
+import matplotlib.pyplot as plt
+
+from src.datasources import floodscan
+
+# Get today's date and save as run_date
+run_date = datetime.datetime.today()
+
+# Extract the year from run_date
+current_year = run_date.year
+
+# Create a sequence of the last 10 years, exclusive of the current year
+last_10_years = list(range(current_year - 10, current_year))
+
+# open historical netcdf file from gdrive just bc so much faster than loading
+# from the blob
+da_fs = floodscan.open_historical_floodscan()
+
+# rolling window to smooth out historical data
+da_fs_smooth = da_fs.rolling(time=10, center=True).mean()
```

Comment:
What do we think about 10 days as the window? I mean this seems about right to me (I've often used 7, partially just because having `center=True` felt nicer on an odd number). I did note that Floodscan already applies some sort of smoothing to their values though. 

I'm generally good with 10, just wondering if there was anything we'd want to check before moving ahead with this number.

Reply:
hmm thats a good point on the `odd` vs `even` number. It's partly a relic of my R-code where the `{terra}` rolling raster stat function `terra::roll(n=10, type = "around")` actually means date +/- 5 days on each side.  However, if that's not happens with `.rolling(time=10)` here, i think we should switch it to have the same effect. The terra roll is a bit peculiar like that as `zoo::roll()` behaves more like you are implying.

Reply:
Interesting- the `xr` `rolling` actually takes an asymmetric window when the size is an even number (so with 10, it takes the current day, 4 ahead, and 5 behind). So yeah if we have `center=True`, we should always use an odd number. 

Another option is to not have it be centered. This would mean that we could report the current date's rolling value on that date, instead of waiting to get the whole window (this is how the app is working). 

So basically our options are:
- centered: any odd number of days (e.g. 7, 11)
- not centered (i.e. only looking backward): any number of days

Reply:
yeah that makes sense its also how `zoo::rollmean()` works in R - `terra::(roll, type = "around")` is the odd ball. 

I changed to 11 and kept centered to be consistent with the R work thats already been presented/'endorsed' - [1aa237a](https://github.com/OCHA-DAP/hdx-floodscan/pull/1/commits/1aa237a03e99b08e416fc3c38f9415f45d2db0d4)